File: spring-boot-rest-integrated-demo/src/main/java/com/github/jccode/springbootrestintegrateddemo/form/UserForm.java
package com.github.jccode.springbootrestintegrateddemo.form;

import javax.validation.constraints.Min;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;

public class UserForm {

    @NotNull
    @Size(min = 3, max = 30)
    private String name;

    @NotNull
    @Size(min = 6, max = 20)
    private String password;

    @NotNull
    @Min(18)
    private Integer age;

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public Integer getAge() {
        return age;
    }

    public void setAge(Integer age) {
        this.age = age;
    }
}

File: spring-boot-rest-integrated-demo/src/main/scala/com/github/jccode/springbootrestintegrateddemo/form/UserForm.scala
package com.github.jccode.springbootrestintegrateddemo.form

import javax.validation.constraints.{Min, NotNull, Size}

import scala.beans.BeanProperty

case class UserForm(@BeanProperty @NotNull @Size(min = 3, max = 30) var name: String,
                    @BeanProperty @NotNull @Size(min = 6, max = 20) var password: String,
                    @BeanProperty @NotNull @Min(18) var age: Int) {
}
